src/workerd/server/workerd-api.c++

```diff
@@ -354,10 +398,59 @@ void WorkerdApi::compileModules(
   lockParam.withinHandleScope([&] {
     auto modules = jsg::ModuleRegistryImpl<JsgWorkerdIsolate_TypeWrapper>::from(lockParam);
 
+    if (util::Autogate::isEnabled(util::AutogateKey::BUILTIN_WASM_MODULES) &&
+        conf.getModules().size() > 0 &&
```

Comment:
Nitpick: I think this line is redundant now. (If there are no modules, `hasPythonModules()` will of course return empty.)

Aside: With Cap'n Proto, getter methods for nontrivial types have a cost, since they have to validate and interpret the pointer, as well as count it against the traversal limit. Here, you have three lines in a row invoking `conf.getModules()`, re-running the same validation. It's a good to form a habit of factoring these out into a local variable, though admittedly in this particular case it's not likely to matter.

Reply:
Good call outs. I'll keep that in mind.

FWIW I prefer to check the size explicitly before performing operations that depend on it. You're right that `hasPythonModules` implicitly checks for the same, it's definitely unlikely in this case but in general implementations change so I prefer to do explicit checks.

Reply:
I changed this in any case.